airflow/providers/amazon/provider.yaml

```diff
@@ -118,10 +118,7 @@ additional-extras:
 devel-dependencies:
   - aiobotocore>=2.7.0
   - aws_xray_sdk>=2.12.0
-  # Moto 5 replaced all decorators with single mock_aws decorator and we need to
-  # Replace the usage of mock decorators in our tests to be able to use moto 5
-  # See https://github.com/apache/airflow/issues/37053
-  - moto[cloudformation,glue]>=4.2.12,<5.0.0
+  - moto[cloudformation,glue]>=5.0.0
```

Comment:
I assume it is not possible to support both 4 and 5? 🥲

Reply:
I think that technically it is possible to make it work with both version for make it possible it required to create a workaround function/decorator/context manager which return appropriate decorator/context manager based on the version of `moto`, however due to the fact that this is devel only dependency and affect only tests so maybe it not worthwhile to spend time to create this workaround and just bump to v5

Reply:
Agree with @Taragolis . With devel-only dependencies it's perfectly fine to "move fast/break things". Since we are fully in control of the environment and we make sure that it is correct and we even expect our contributors in main to have exactly the same set of dependencies as everyone else. This is why we have constraints, so generally we should be ok even if with every single commit we add `>= <current constraint version of the dependency>`. 

That would be even good as it would force anyone to get the `proper` versions of their development environment which is "same as everyone else" - but yeah, that would be an exaggeration.

In the new PEP-compliant world and our hatchling backend, those `devel-dependencies` are even stripped out automatically  (by our hatchling build hooks) from the "user-facing" wheel, so it has exactly 0 impact on users, it only affects those who contribute to airflow.